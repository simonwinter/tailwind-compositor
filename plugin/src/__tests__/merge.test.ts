import transform from '../theme-transform';

import compositorConfig from './stubs/compositor.config.js';
import tailwindConfig from './stubs/tailwind.config.js';

const transformer = transform(compositorConfig);

test('runs', () => {
	const tailwindTransformed = transformer(tailwindConfig);
	expect(tailwindTransformed).toMatchSnapshot();
});
