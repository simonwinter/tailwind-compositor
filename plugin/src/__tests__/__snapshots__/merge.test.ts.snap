// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runs 1`] = `
Object {
  "corePlugins": false,
  "important": false,
  "plugins": Array [
    [Function],
  ],
  "prefix": "",
  "purge": Array [],
  "separator": ":",
  "target": "relaxed",
  "theme": Object {
    "colors": Object {
      "black": "#000",
      "current": "currentColor",
      "transparent": "transparent",
      "white": "#fff",
    },
    "compositor": Object {
      "baseline": 8,
      "fonts": Array [
        Object {
          "ascent": 1025,
          "capHeight": 698,
          "descent": -275,
          "fallback": "sans-serif",
          "familyName": "IBM Plex Sans",
          "italic": false,
          "key": "sans-400",
          "upm": 1000,
          "weight": 400,
          "xHeight": 525,
        },
      ],
      "measure": Array [
        15,
        20,
        25,
        50,
        55,
        60,
        65,
        70,
      ],
      "rhythm": Array [
        0,
        1,
        2,
        3,
        4,
        5,
        6,
        8,
        10,
        12,
        16,
        21,
        24,
        28,
      ],
      "root": 16,
      "type": Array [
        14,
        16,
        18,
        20,
        22,
        24,
        28,
        32,
        36,
        40,
      ],
      "useRem": true,
    },
    "extend": Object {
      "height": Object {
        "0": "0rem",
        "1": "0.5rem",
        "10": "8rem",
        "11": "10.5rem",
        "12": "12rem",
        "13": "14rem",
        "2": "1rem",
        "3": "1.5rem",
        "4": "2rem",
        "5": "2.5rem",
        "6": "3rem",
        "7": "4rem",
        "8": "5rem",
        "9": "6rem",
      },
      "maxHeight": Object {
        "0": "0rem",
        "1": "0.5rem",
        "10": "8rem",
        "11": "10.5rem",
        "12": "12rem",
        "13": "14rem",
        "2": "1rem",
        "3": "1.5rem",
        "4": "2rem",
        "5": "2.5rem",
        "6": "3rem",
        "7": "4rem",
        "8": "5rem",
        "9": "6rem",
        "extend": "100px",
      },
      "maxWidth": Object {
        "extend": "100px",
      },
      "minHeight": Object {
        "0": "0rem",
        "1": "0.5rem",
        "10": "8rem",
        "11": "10.5rem",
        "12": "12rem",
        "13": "14rem",
        "2": "1rem",
        "3": "1.5rem",
        "4": "2rem",
        "5": "2.5rem",
        "6": "3rem",
        "7": "4rem",
        "8": "5rem",
        "9": "6rem",
        "extend": "100px",
      },
      "minWidth": Object {
        "extend": "100px",
      },
    },
    "flex": Object {
      "auto": "1 1 auto",
    },
    "fontFamily": Object {
      "sans": Array [
        "system-ui",
      ],
    },
    "fontSize": Object {
      "0": "0.875rem",
      "1": "1rem",
      "2": "1.125rem",
      "3": "1.25rem",
      "4": "1.375rem",
      "5": "1.5rem",
      "6": "1.75rem",
      "7": "2rem",
      "8": "2.25rem",
      "9": "2.5rem",
    },
    "fontWeight": Object {
      "bold": "700",
      "normal": "400",
    },
    "letterSpacing": Object {
      "normal": "0",
      "wide": "0.025em",
    },
    "lineHeight": Object {
      "none": "1",
      "normal": "1.5",
    },
    "screens": Object {
      "sm": "640px",
    },
    "spacing": Object {
      "0": "0rem",
      "1": "0.5rem",
      "10": "8rem",
      "11": "10.5rem",
      "12": "12rem",
      "13": "14rem",
      "2": "1rem",
      "3": "1.5rem",
      "4": "2rem",
      "5": "2.5rem",
      "6": "3rem",
      "7": "4rem",
      "8": "5rem",
      "9": "6rem",
    },
  },
  "variants": Object {
    "accessibility": Array [],
    "alignContent": Array [],
    "alignItems": Array [],
    "alignSelf": Array [],
    "appearance": Array [],
    "backgroundAttachment": Array [],
    "backgroundColor": Array [],
    "backgroundOpacity": Array [],
    "backgroundPosition": Array [],
    "backgroundRepeat": Array [],
    "backgroundSize": Array [],
    "borderCollapse": Array [],
    "borderColor": Array [],
    "borderOpacity": Array [],
    "borderRadius": Array [],
    "borderStyle": Array [],
    "borderWidth": Array [],
    "boxShadow": Array [],
    "boxSizing": Array [],
    "clear": Array [],
    "cursor": Array [],
    "display": Array [],
    "divideColor": Array [],
    "divideOpacity": Array [],
    "divideWidth": Array [],
    "fill": Array [],
    "flex": Array [],
    "flexDirection": Array [],
    "flexGrow": Array [],
    "flexShrink": Array [],
    "flexWrap": Array [],
    "float": Array [],
    "fontFamily": Array [],
    "fontSize": Array [],
    "fontSmoothing": Array [],
    "fontStyle": Array [],
    "fontWeight": Array [],
    "gap": Array [],
    "gridAutoFlow": Array [],
    "gridColumn": Array [],
    "gridColumnEnd": Array [],
    "gridColumnStart": Array [],
    "gridRow": Array [],
    "gridRowEnd": Array [],
    "gridRowStart": Array [],
    "gridTemplateColumns": Array [],
    "gridTemplateRows": Array [],
    "height": Array [],
    "inset": Array [],
    "justifyContent": Array [],
    "letterSpacing": Array [],
    "lineHeight": Array [],
    "listStylePosition": Array [],
    "listStyleType": Array [],
    "margin": Array [],
    "maxHeight": Array [],
    "maxWidth": Array [],
    "minHeight": Array [],
    "minWidth": Array [],
    "objectFit": Array [],
    "objectPosition": Array [],
    "opacity": Array [],
    "order": Array [],
    "outline": Array [],
    "overflow": Array [],
    "padding": Array [],
    "placeholderColor": Array [],
    "placeholderOpacity": Array [],
    "pointerEvents": Array [],
    "position": Array [],
    "resize": Array [],
    "rotate": Array [],
    "scale": Array [],
    "skew": Array [],
    "space": Array [],
    "stroke": Array [],
    "strokeWidth": Array [],
    "tableLayout": Array [],
    "textAlign": Array [],
    "textColor": Array [],
    "textDecoration": Array [],
    "textOpacity": Array [],
    "textTransform": Array [],
    "transform": Array [],
    "transformOrigin": Array [],
    "transitionDelay": Array [],
    "transitionDuration": Array [],
    "transitionProperty": Array [],
    "transitionTimingFunction": Array [],
    "translate": Array [],
    "userSelect": Array [],
    "verticalAlign": Array [],
    "visibility": Array [],
    "whitespace": Array [],
    "width": Array [],
    "wordBreak": Array [],
    "zIndex": Array [],
  },
}
`;
